File: local_examples/cmds_cnxmgmt/seredis/CmdsCnxmgmtExample.cs
// EXAMPLE: cmds_cnxmgmt
using System;
using StackExchange.Redis;

public class CmdsCnxmgmt
{
    public void Run()
    {
        var muxer = ConnectionMultiplexer.Connect("localhost:6379");
        var db = muxer.GetDatabase();

        // Note: StackExchange.Redis has no dedicated AUTH method — you normally
        // supply credentials on the connection (for example
        // ConfigurationOptions.User / ConfigurationOptions.Password). The raw
        // AUTH command is shown here for parity with the other clients.

        // STEP_START auth1
        var res1 = db.Execute("AUTH", "temp_pass");
        Console.WriteLine(res1); // >>> OK

        var res2 = db.Execute("AUTH", "default", "temp_pass");
        Console.WriteLine(res2); // >>> OK
        // STEP_END

        // STEP_START auth2
        var res3 = db.Execute("AUTH", "test-user", "strong_password");
        Console.WriteLine(res3); // >>> OK
        // STEP_END
    }
}

File: local_examples/cmds_cnxmgmt/lettuce-reactive/CmdsCnxmgmtExample.java
// EXAMPLE: cmds_cnxmgmt
package io.redis.examples.reactive;

import io.lettuce.core.RedisClient;
import io.lettuce.core.api.reactive.RedisReactiveCommands;
import io.lettuce.core.api.StatefulRedisConnection;
// REMOVE_START
import org.junit.jupiter.api.Test;
// REMOVE_END

import reactor.core.publisher.Mono;

public class CmdsCnxmgmtExample {

    // REMOVE_START
    @Test
    // REMOVE_END
    public void run() {
        RedisClient redisClient = RedisClient.create("redis://localhost:6379");

        try (StatefulRedisConnection<String, String> connection = redisClient.connect()) {
            RedisReactiveCommands<String, String> reactiveCommands = connection.reactive();

            // STEP_START auth1
            Mono<Void> authExample1 = reactiveCommands.auth("temp_pass")
                    .doOnNext(res1 -> {
                        System.out.println(res1); // >>> OK
                    })
                    .then(reactiveCommands.auth("default", "temp_pass"))
                    .doOnNext(res2 -> {
                        System.out.println(res2); // >>> OK
                    })
                    .then();
            // STEP_END

            authExample1.block();

            // STEP_START auth2
            Mono<Void> authExample2 = reactiveCommands.auth("test-user", "strong_password")
                    .doOnNext(res3 -> {
                        System.out.println(res3); // >>> OK
                    })
                    .then();
            // STEP_END

            authExample2.block();
        } finally {
            redisClient.shutdown();
        }
    }
}
